### PR TITLE
Add Facebook login

### PR DESCRIPTION
**Pull request by wolfd**
Adds Facebook login button and integrates it with Firebase

### CODE REVIEW

File: app/src/main/java/pizza/olin/consamables/LoginActivity.java [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+                                    Toast.LENGTH_SHORT).show();
+                        }
+
+                        // ...
+                    }
+                });
+    }
+
+    private void logout() {
+        FirebaseAuth.getInstance().signOut();
```
Comment (sammyers) [inline app/src/main/java/pizza/olin/consamables/LoginActivity.java:124]: Can this be `mAuth.signOut()`?

File: app/src/main/java/pizza/olin/consamables/LoginActivity.java [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+        LoginButton loginButton = (LoginButton) findViewById(R.id.login_button);
+        loginButton.setReadPermissions("email", "public_profile");
+        // Other app specific specialization
+
+        // Callback registration
+        loginButton.registerCallback(callbackManager, new FacebookCallback<LoginResult>() {
+            @Override
+            public void onSuccess(LoginResult loginResult) {
+                // App code
```
Comment (sammyers) [inline app/src/main/java/pizza/olin/consamables/LoginActivity.java:69]: Would be good to know what you plan to do here.

**Review by sammyers — COMMENTED** [review]
Overall looks good. Better scaffolding of functionality in between boilerplate would be solid, and a couple of nitpicky details.

File: app/src/main/java/pizza/olin/consamables/LoginActivity.java (hunk 2 above)
Comment (wolfd) [reply app/src/main/java/pizza/olin/consamables/LoginActivity.java:69]: Nothing yet. I was going to wait for a second pass before coming back to this.

File: app/src/main/java/pizza/olin/consamables/LoginActivity.java (hunk 1 above)
Comment (wolfd) [reply app/src/main/java/pizza/olin/consamables/LoginActivity.java:124]: Yes, this is more of a note for the time being than a permanent block of code.

**Comment by wolfd** [comment]
This is a first pass model, I'm going to come back and fix it up when it's integrated with the rest of the app.

**Pull request merged**